src/main/java/address/updater/DependencyHistoryHandler.java

```diff
@@ -0,0 +1,118 @@
+package address.updater;
+
+import address.MainApp;
+import address.storage.StorageManager;
+import address.util.*;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.util.*;
+import java.util.jar.Attributes;
+import java.util.jar.Manifest;
+import java.util.stream.Collectors;
+
+/**
+ * Tracks dependencies used by current version and backup versions of the application
+ * Cleans up dependencies not used by mentioned versions
+ */
+public class DependencyHistoryHandler {
+    private static final AppLogger logger = LoggerManager.getLogger(DependencyHistoryHandler.class);
+    private static final File DEPENDENCY_HISTORY_FILE = new File("lib/dependency_history");
+
+    private final Version currentVersion;
+    private HashMap<Version, List<String>> dependenciesForVersionsInUse = new HashMap<>();
+
+    public DependencyHistoryHandler(Version currentVersion) {
+        this.currentVersion = currentVersion;
+
+        loadVersionDependencyHistory();
+
+        if (!ManifestFileReader.isRunFromJar()) {
+            logger.debug("Not running from JAR, will not update version dependencies");
+            return;
+        }
+
+        Optional<List<String>> libraries = ManifestFileReader.getLibrariesInClasspathFromManifest();
+
+        if (!libraries.isPresent()) {
+            logger.debug("No libraries being used - should not happen");
+            assert false;
+            return;
+        }
```

Comment:
This whole block can be reduced to this?
`assert libraries.isPresent() : "No libraries being used - should not happen";`